cdm/core/src/main/java/ucar/nc2/filter/Filters.java

```diff
@@ -10,38 +10,95 @@
 
 public class Filters {
 
+  /**
+   * Set of common properties used by Filters
+   */
+  public class Keys {
+    /**
+     * key for a String id/name (e.g. Zarr filters)
+     */
+    public static final String NAME = "id";
+    /**
+     * key for a numeric id (e.g. HDF5 filters)
+     */
+    public static final String ID = "nid";
+    /**
+     * key for property indicating whether filter is optional,
+     * i.e. whether it should continue or throw if it fails
+     */
+    public static final String OPTIONAL = "optional";
+    /**
+     * key mapping to client data used by filter
+     */
+    public static final String DATA = "data";
+    /**
+     * key mapping to element size
+     */
+    public static final String ELEM_SIZE = "elementsize";
+  }
+
   private static NullFilter nullFilter = new NullFilter();
 
   /**
-   * Create a filter object for the given property list
+   * Create a filter object matching the provided id
    * 
    * @param properties
-   * @return A filter object with the provided properties
+   * @return A filter object with the given properties
    * @throws UnknownFilterException
    */
-  public static Filter getFilterByName(Map<String, Object> properties) throws UnknownFilterException {
+  public static Filter getFilter(Map<String, Object> properties) throws UnknownFilterException {
     if (properties == null) {
       return nullFilter;
     }
-    String name = (String) properties.get("id");
-    if (name == null || name.isEmpty()) {
+    // read id properties
+    String name = (String) properties.get(Keys.NAME);
+    Object oid = properties.get(Keys.ID);
+    // if no id or name, return null filter
+    if ((name == null || name.isEmpty()) && (oid == null || !(oid instanceof Number))) {
```

Comment:
second condition could be simplified
```suggestion
    if ((name == null || name.isEmpty()) && !(oid instanceof Number)) {
```